dowhy/causal_estimators/econml.py

```diff
@@ -20,21 +22,88 @@ class Econml(CausalEstimator):
 
     """
 
-    def __init__(self, *args, econml_methodname, **kwargs):
+    def __init__(
+        self,
+        identified_estimand: IdentifiedEstimand,
+        econml_methodname: str,
```

Comment:
Could this already be a `Union[str,EconMLEstimator]` where `EconMLEstimator` is something along the lines of:
```python
class EconMLEstimator(Protocol):
    def estimate(self, ...):
        ...

    ...
```

Then, when actually using this, you could check if it's a string or not. Long-term, we could deprecate and remove the string and just allow the estimator objects themselves. That would make usage and implementation simpler.

Reply:
+1 This is a good idea to maintain backwards compatibility while still following the new API.

Reply:
Yep, I actually created a sample of this to show Amit some days ago :) I think we could even deprecate the string now and move the code that creates an econml instance from the string to the CausalModel estimate_effect class, what do you think? @petergtz

Reply:
> we could even deprecate the string now and move the code that creates an econml instance from the string to the CausalModel estimate_effect class

From a point of getting more concrete on this, I like it. But in terms of backwards compatibility, I'm not sure sure we should already be so bold. But that's mostly depending on the question of "is this code used directly by users" or will this only be invoked through estimate_effect. @amit-sharma ?

Either way, we could think about raising a `DeprecationWarning` already when a string is passed.